clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10159,13 +10159,24 @@ void Cgeist::ConstructJob(Compilation &C, const JobAction &JA,
 
   TCArgs.AddAllArgValues(CmdArgs, options::OPT_Xcgeist);
 
+  const InputInfo *HostModule = nullptr;
   for (const InputInfo &I : Inputs) {
-    if (I.isFilename())
+    if (I.getType() == types::TY_MLIR_IR)
```

Comment:
```suggestion
    if (I.getType() == types::TY_MLIR_IR && I.Act->isHostOffloading(Action::OFK_SYCL))
```

Nitpick, As this should map to the LLVM IR -> MLIR compile action, that should capture this is actually the host (and not a device MLIR)